beluga/test/beluga/sensor/test_lfm_with_unknown_space.cpp

```diff
@@ -0,0 +1,271 @@
+    -1 , -1 , -1 , 100, 100,
+    -1 , 0  , 0  , 0  , 100,
+    -1 , 0  , 0  , 0  , 100,
+    100, 0  , 0  , 0  , 100,
+    100, 100, 100, 100, 100},
+    kResolution};
+  // clang-format on
+
+  const double param_mld = 20.0;
+  const double unknown_likelihood = 1 / param_mld;
+  const auto params = beluga::LikelihoodFieldModelParam{2.0, param_mld, 0.5, 0.5, 0.2, true};
+  auto sensor_model = UUT{params, grid};
+
+  // clang-format off
+ const double expected_likelihood_field[] = {  // NOLINT(modernize-avoid-c-arrays)
+    unknown_likelihood, unknown_likelihood, unknown_likelihood, 1.022, 1.022,
+    unknown_likelihood, 0.025             , 0.027             , 0.069, 1.022,
+    unknown_likelihood, 0.027             , 0.025             , 0.069, 1.022,
+    1.022             , 0.069             , 0.069             , 0.069, 1.022,
+    1.022             , 1.022             , 1.022             , 1.022, 1.022
+  };
+  // clang-format on
+
+  // the likelihood field model includes an optimization that stores the likelihood elevated to the cube
+  auto expected_cubed_likelihood =
+      expected_likelihood_field | ranges::views::transform([](auto v) { return v * v * v; }) | ranges::to<std::vector>;
+
+  ASSERT_THAT(
+      sensor_model.likelihood_field().data(),
+      testing::Pointwise(testing::DoubleNear(0.003), expected_cubed_likelihood));
+}
+
+TEST(LikelihoodFieldModelUnknownSpace, LikelihoodField2) {
+  constexpr double kResolution = 0.5;
+  // clang-format off
+  const auto grid = GeneralOccupancyGrid<5, 5>{{
+    -1, -1, -1, 100, 100,
+    -1, -1, -1, 0  , 0  ,
+    -1, -1, -1, 0  , 0  ,
+    -1, -1, -1, 0  , 0  ,
+    -1, -1, -1, 100, 100},
+    kResolution};
+  // clang-format on
+
+  const double param_mld = 100.0;
+  const double unknown_likelihood = 1 / param_mld;
+  const auto params = beluga::LikelihoodFieldModelParam{2.0, param_mld, 0.5, 0.5, 0.2, true};
+  auto sensor_model = UUT{params, grid};
+
+  // clang-format off
+  const double expected_likelihood_field[] = {  // NOLINT(modernize-avoid-c-arrays)
+    unknown_likelihood, unknown_likelihood, unknown_likelihood, 1.002, 1.002,
+    unknown_likelihood, unknown_likelihood, unknown_likelihood, 0.049, 0.049,
+    unknown_likelihood, unknown_likelihood, unknown_likelihood, 0.005, 0.005,
+    unknown_likelihood, unknown_likelihood, unknown_likelihood, 0.049, 0.049,
+    unknown_likelihood, unknown_likelihood, unknown_likelihood, 1.002, 1.002,
+  };
+  // clang-format on
+
+  // the likelihood field model includes an optimization that stores the likelihood elevated to the cube
+  auto expected_cubed_likelihood =
+      expected_likelihood_field | ranges::views::transform([](auto v) { return v * v * v; }) | ranges::to<std::vector>;
+
+  ASSERT_THAT(
+      sensor_model.likelihood_field().data(),
+      testing::Pointwise(testing::DoubleNear(0.003), expected_cubed_likelihood));
+}
+
+TEST(LikelihoodFieldModelUnknownSpace, ImportanceWeight) {
```

Comment:
@DPR00 it would be nice to have a test with a 1-point scan that hits an unknown cell, to show it in fact returns `kUnknownSpaceLikelihood`.

Reply:
I add the following tests:
```cpp
  {
    auto state_weighting_function = sensor_model(std::vector<std::pair<double, double>>{{0, 0}});
    ASSERT_NEAR(
        1.000 + kUnknownSpaceLikelihood * kUnknownSpaceLikelihood * kUnknownSpaceLikelihood,
        state_weighting_function(grid.origin()), 0.003);
  }

  {
    auto state_weighting_function = sensor_model(std::vector<std::pair<double, double>>{{2.25, 2.25}});
    ASSERT_NEAR(
        1.000 + kUnknownSpaceLikelihood * kUnknownSpaceLikelihood * kUnknownSpaceLikelihood,
        state_weighting_function(grid.origin()), 0.003);
  }
```
Keep in mind that `kUnknownSpaceLikelihood * kUnknownSpaceLikelihood * kUnknownSpaceLikelihood` is neglictible compared to `1.000`, that is why I had written `1.000` previously.